gno.land/pkg/sdk/vm/handler.go

```diff
@@ -64,21 +77,23 @@ func (vh vmHandler) handleMsgRun(ctx sdk.Context, msg MsgRun) (res sdk.Result) {
 	if err != nil {
 		return abciResult(err)
 	}
+
 	res.Data = []byte(resstr)
-	return
+	return res
 }
 
 // ----------------------------------------
 // Query
 
 // query paths
 const (
-	QueryPackage = "package"
-	QueryStore   = "store"
-	QueryRender  = "qrender"
-	QueryFuncs   = "qfuncs"
-	QueryEval    = "qeval"
-	QueryFile    = "qfile"
+	QueryPackage  = "package"
+	QueryStore    = "store"
+	QueryRender   = "qrender"
+	QueryFuncs    = "qfuncs"
+	QueryEval     = "qeval"
+	QueryEvalJSON = "qeval/json" // EXPERIMENTAL
```

Comment:
I think I prefer a single call with a flag to specify the format.
Any other opinions?